autoload/airline/extensions/branch.vim

```diff
@@ -73,9 +73,16 @@ function! airline#extensions#branch#get_head()
   let empty_message = get(g:, 'airline#extensions#branch#empty_message',
       \ get(g:, 'airline_branch_empty_message', ''))
   let symbol = get(g:, 'airline#extensions#branch#symbol', g:airline_symbols.branch)
+  let tracked = s:check_file_tracked() ? '' : ' (UNTRACKED FILE)'
   return empty(head)
         \ ? empty_message
-        \ : printf('%s%s', empty(symbol) ? '' : symbol.(g:airline_symbols.space), head)
+        \ : printf('%s%s%s', empty(symbol) ? '' : symbol.(g:airline_symbols.space), head, tracked)
+endfunction
+
+function! s:check_file_tracked()
+  let file = expand("%")
+  let is_tracked = (system('git status ' . file) =~ 'Untracked') ? 0 : 1
```

Comment:
For reference, `git ls-files --other --directory --exclude-standard` is faster.

And if you have GNU sed, you can pipe it to `sed -n q1`, which exits with the first output (in case of larger repos).